### PR TITLE
#27 Fix obj serialization for saving

### LINKED ISSUE #27: Result saving fails because ModelType is not serializable
## Description
When setting the `save_results` attribute to `True` for the evaluation & batch evaluation methods of FlowJudge/AsyncFlowJudge classes, it raises an error and the saving fails

### Expected behaviour
The results are successfully saved.

### Actual behaviour
Raises the following error:
`TypeError: Object of type ModelType is not JSON serializable`

### PR DESCRIPTION
**Pull request by minaamshahid**
Fixes #27

### CODE REVIEW

**Comment by sariola** [comment]
Added stronger, more defensive input checks for the result_writer and unit tests.
Robustness improvements to the file name encoding.

File: flow_judge/utils/result_writer.py [hunk 1]
```diff
@@ -13,55 +17,222 @@
...
+        ValueError: If eval_inputs and eval_outputs have different lengths.
+
+    Note:
+        - Uses UTF-8 encoding.
+        - Overwrites the file if it already exists.
+        - Each line in the file is a JSON object representing one result.
+        - Ensures non-ASCII characters are preserved in the output.
+    """
+    if len(eval_inputs) != len(eval_outputs):
+        raise ValueError("eval_inputs and eval_outputs must have the same length")
```
[the remark this answers is not in the document]
Comment (sariola) [reply flow_judge/utils/result_writer.py:239]: Thanks Minaam, it's the zip function below that needs them to be the same lengths. This check is just to throw the right error type.

I'll see what I could do.

image

**Comment by sariola** [comment]
I've pushed 'append' style result_writing for batches. 

I still need to test a bit more and understand what makes the most sense.

File: flow_judge/flow_judge.py [hunk 2]
```diff
@@ -129,6 +129,46 @@ def __init__(
         if not isinstance(model, AsyncBaseFlowJudgeModel):
             raise ValueError("Invalid model type. Use AsyncBaseFlowJudgeModel or its subclasses.")
 
+    def _handle_batch_result(
+        self, batch_result: BatchResult, batch_len: int, fail_on_parse_error: bool
+    ) -> list[EvalOutput]:
+        """Handle output parsing for batched results from Baseten.
```
Comment (bergr7) [inline flow_judge/flow_judge.py:135]: This is probably for all model instances not just for baseten right?

**Review by sariola — APPROVED** [review]

**Pull request merged**